### DISCUSSION TITLE
Readme reworking

### DISCUSSION THREAD

**Discussion by Ilithy** [TrianguloY/URLCheck#43] [Ideas]
Hi @TrianguloY , I had fun modifying your Readme to make it more visually appealing, 
Here is the link to the branch where I made the modifications:
https://github.com/Ilithy/UrlChecker/tree/readme-rework

(Honestly) Do you like the changes? (This is not extraordinary, I am aware of it), I can of course adapt to your tastes, this is only a first draft (and this is only a proposal, nothing mandatory ^_°)

**Comment by TrianguloY** [comment]
Wow, it does even look professional! Nice work!

Some comments though:
- The top image could be a nice replacement for the [featured](https://github.com/Ilithy/UrlChecker/blob/readme-rework/app/src/main/play/listings/en-US/graphics/feature-graphic/featured.png) one, although the text is a bit hard to read (the background graphs merge with it). Maybe add a small white fading around the text? Also, where did you got that background? I like it but I want to make sure its free to use.
- What about placing the 'download' section between the 'Android app made by TrianguloY' and 'Url checker acts as an intermediary...' this way you won't need to inline link either. On that section I like to have the link as text too, that may be personal preference.
- Placing the screenshots there is a wonderful idea, although I think it's better to just link the [existing ones](https://github.com/Ilithy/UrlChecker/tree/readme-rework/app/src/main/play/listings/en-US/graphics/phone-screenshots) so that they will automatically update. Replacing them is also needed, that's for another day.
- Nice touch adding links on the contribution actions [the 'contributions' title is smaller though]

About the assets, instead of adding them to a new 'art' folder I think it's better to just link the existing ones from the [graphics folder](https://github.com/Ilithy/UrlChecker/tree/readme-rework/app/src/main/play/listings/en-US/graphics). The creative commons icon can be linked using the [official badge](https://creativecommons.org/choose/results-one?license_code=by&amp;jurisdiction=&amp;version=4.0&amp;lang=en)

Over all It's a really nice improvement and I would like to update it, thanks!

**Reply by Ilithy** [reply]
>Wow, it does even look professional! Nice work!

Thanks but it's not much, I'm cheating a bit.

>The top image could be a nice replacement for the [featured](https://github.com/Ilithy/UrlChecker/blob/readme-rework/app/src/main/play/listings/en-US/graphics/feature-graphic/featured.png) one,

Yes, when a version will suit you, I will do the replacement of the files (instead of leaving it in the art/ folder).

>although the text is a bit hard to read (the background graphs merge with it). Maybe add a small white fading around the text?

I understand, I will adapt it according to that, here are two proposals,

![presentation banner(1)](https://user-images.githubusercontent.com/36798218/183245026-e800af42-fdd2-45ae-b64e-28657558a26e.jpg)

_with a little color (but just a little ;))_
![presentation banner](https://user-images.githubusercontent.com/36798218/183245033-b8ea3247-ac8b-4e29-8b86-cf1735eeea9a.jpg)

does it come close to what you want?

> I like it but I want to make sure its free to use.

Yes it is free, the author just wants to be credited (Image of presentation: Garry Killian) I just modified it a little.
Here is a link to the original image: https://www.freepik.com/free-vector/abstract-mesh-background_20618934.htm?query=link

>What about placing the 'download' section between the 'Android app made by TrianguloY' and 'Url checker acts as an intermediary...' this way you won't need to inline link either.

these modified, but I'm not sure if this version is the right one (my current idea, not your comment), you'll have to give me your opinion, thanks.

>On that section I like to have the link as text too, that may be personal preference.

I admit, I don't find it elegant, but it's personal, and it's your application, I will modify this

>Placing the screenshots there is a wonderful idea, although I think it's better to just link the [existing ones](https://github.com/Ilithy/UrlChecker/tree/readme-rework/app/src/main/play/listings/en-US/graphics/phone-screenshots) so that they will automatically update.

Just to make sure I understand, do you want me to place the newly added "screenshots" (0.png, 1.png etc...) in the listings/en-US/graphics/phone-screenshots folder in addition to the old ones? instead of the old ones? or do you want me to use the items in the current phone-screenshots folder for the readme, I'm not sure.

>Replacing them is also needed, that's for another day.

I was also planning to update the app store screenshots later, with the same type (screenshot surrounded on a transparent background), it looks very good on F-droid and Google Play store I think, what do you think?

>Nice touch adding links on the contribution actions [the 'contributions' title is smaller though]

Thank you, if the size matters (^^), I will return it to its original size

>About the assets, instead of adding them to a new 'art' folder I think it's better to just link the existing ones from the [graphics folder](https://github.com/Ilithy/UrlChecker/tree/readme-rework/app/src/main/play/listings/en-US/graphics).

I'll take care of it.

>The creative commons icon can be linked using the [official badge](https://creativecommons.org/choose/results-one?license_code=by&jurisdiction=&version=4.0&lang=en)

Yes indeed, but it is less pretty (this modification is already in place, tell me if it suits you that way)

I will make the others changes this afternoon.

**Reply by TrianguloY** [reply]
> > The top image could be a nice replacement for the [featured](https://github.com/Ilithy/UrlChecker/blob/readme-rework/app/src/main/play/listings/en-US/graphics/feature-graphic/featured.png) one,
> 
> Yes, when a version will suit you, I will do the replacement of the files (instead of leaving it in the art/ folder).

No need to wait, it's your branch after all, just replace the file and use it directly. If necessary I'll take care of replace it before merging.

> > although the text is a bit hard to read (the background graphs merge with it). Maybe add a small white fading around the text?
> 
> I understand, I will adapt it according to that, here are two proposals,
> 
> ![presentation banner(1)]
> 
> _with a little color (but just a little ;))_ ![presentation banner]
> 
> does it come close to what you want?

Hmm, I was thinking of something different, something like this:
![image](https://user-images.githubusercontent.com/10616202/183247958-b7d34d0e-093a-4c10-8f8d-10423547a1db.png)

Btw, play store is very strict about the image sizes, the store says: `Your feature graphic must be a PNG or JPEG, up to 1MB, and 1,024 px by 500 px`


> > I like it but I want to make sure its free to use.
> 
> Yes it is free, the author just wants to be credited (Image of presentation: Garry Killian) I just modified it a little. Here is a link to the original image: https://www.freepik.com/free-vector/abstract-mesh-background_20618934.htm?query=link

Ok, I think there is space for the credit in the readme and in the play store description.

> > What about placing the 'download' section between the 'Android app made by TrianguloY' and 'Url checker acts as an intermediary...' this way you won't need to inline link either.
> 
> these modified, but I'm not sure if this version is the right one (my current idea, not your comment), you'll have to give me your opinion, thanks.

Now it's ok!

> > On that section I like to have the link as text too, that may be personal preference.
> 
> I admit, I don't find it elegant, but it's personal, and it's your application, I will modify this

None of the things I tried look good, so I think it's better to remove the links, yes. Just center them and remove the inline from `Url Checker acts as an intermediary...`

> > Placing the screenshots there is a wonderful idea, although I think it's better to just link the [existing ones](https://github.com/Ilithy/UrlChecker/tree/readme-rework/app/src/main/play/listings/en-US/graphics/phone-screenshots) so that they will automatically update.
> 
> Just to make sure I understand, do you want me to place the newly added "screenshots" (0.png, 1.png etc...) in the listings/en-US/graphics/phone-screenshots folder in addition to the old ones? instead of the old ones? or do you want me to use the items in the current phone-screenshots folder for the readme, I'm not sure.
>
> > Replacing them is also needed, that's for another day.
> 
> I was also planning to update the app store screenshots later, with the same type (screenshot surrounded on a transparent background), it looks very good on F-droid and Google Play store I think, what do you think?

Replace the screenshots in listings/en-US/graphics/phone-screenshots with your, and link them. We can discuss the exact screenshots later if needed, but linking them allows the readme to by up-to-date without duplicates. I like your screenshots too.

> > About the assets, instead of adding them to a new 'art' folder I think it's better to just link the existing ones from the [graphics folder](https://github.com/Ilithy/UrlChecker/tree/readme-rework/app/src/main/play/listings/en-US/graphics).
> 
> I'll take care of it.

As explained above, just replace the images with yours (it's your branch, you can do anything there) and when merging I'll make sure to replace them if needed. Basically remove the art folder.

> > The creative commons icon can be linked using the [official badge](https://creativecommons.org/choose/results-one?license_code=by&jurisdiction=&version=4.0&lang=en)
> 
> Yes indeed, but it is less pretty (this modification is already in place, tell me if it suits you that way)

That's it! What about centering it?

> Addition of nice little badge (maintainer, donation etc...)

Hey! those look nice, but they seem hardcoded. Aren't there a dynamic mantainance badge? (oh, and the app is made in Java not Javascript ;)

**Comment by Ilithy** [comment]
Summary of (light) changes made:
- The title of the contribution section is at its first size.
- Moved the "download to app store" badges between "Android app made by TrianguloY" and the introductory text
- The links to the app store in text version have been reintroduced in a (hidden) section titled Links (below the download badges)
- Addition of nice little badge (maintainer, donation etc...)
- Very slight modifications of the texts in the contribution section

**Comment by TrianguloY** [comment]
This discussion is being a bit hard to follow. Make a PR and we'll discuss the code there line by line.

Continued on https://github.com/TrianguloY/UrlChecker/pull/44

**Reply by Ilithy** [reply]
Yes, a little, i made the PR
